pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -186,18 +182,24 @@ func filter(context context.Context, schedulerExtenderArgs schedulerapi.Extender
 		excludeNode := false
 		if found && ready {
 			// get AzVolumeAttachments for the node
-			azVolumeAttachments, err := getAdditionalAzVolumeAttachmentsOnNode(node.Name, pvNamesForRequestedVolumes)
+			azVolumeAttachments, err := getAdditionalAzVolumeAttachmentsOnNode(node.Name, diskNamesForRequestedVolumes)
 			klog.V(2).Infof(
 				"number of additional volume attachments for node %s and pod %s is %d. Number of requested volumes is %d. Err: %v",
-				node.Name, schedulerExtenderArgs.Pod.Name, len(azVolumeAttachments), numberOfRequestedAttachments, err)
-			if err == nil {
-				// get the max disk count for the node type
-				maxDiskCount, err := azureutils.GetNodeMaxDiskCount(node.Labels)
-				if err == nil && len(azVolumeAttachments)+numberOfRequestedAttachments > maxDiskCount {
-					klog.Warningf("max number of azVolumeAttachments for node %s has been reached. Node will be skipped.", node.Name)
-					excludeNode = true
-					failedNodes[node.Name] = "node(s) reached the maximum number of disk attachments"
-				}
+				node.Name, schedulerExtenderArgs.Pod.Name, len(azVolumeAttachments), volumeCount, err)
+			if err != nil {
```

Comment:
np: I think these `if`-s and `continue`-s make the code unnecessarily long and harder to follow. 
We could potentially add logging for the `err` of `GetNodeMaxDiskCount` as well.

Reply:
I am for the use of `continue`-s in this case as it prevents us from unnecessary indentation of the codes.

I will go ahead and add logging 👍